src/databricks/labs/ucx/hive_metastore/tables.py

```diff
@@ -143,3 +150,46 @@ def _describe(self, catalog: str, database: str, table: str) -> Table | None:
         except Exception as e:
             logger.error(f"Couldn't fetch information for table {full_name} : {e}")
             return None
+
+    def migrate_tables(self):
+        if self._workspace_config.database_to_catalog_mapping:
+            for table in self._fetch_tables():
+                target_catalog = self._workspace_config.database_to_catalog_mapping.get(
+                    table.database, self._workspace_config.default_catalog
+                )
+                self._migrate_table(target_catalog, table)
+        else:
+            for table in self._fetch_tables():
+                self._migrate_table(self._workspace_config.default_catalog, table)
```

Comment:
```suggestion
        for table in self.snapshot():
                target_catalog = self._database_to_catalog_mapping.get(table.database, self._workspace_config.default_catalog)
                self._migrate_table(target_catalog, table)
```

we don't need two separate branches.

and use `.snapshot()`, because it's the method already doing tables fetch.

also, please put `self._database_to_catalog_mapping.get(table.database, self._workspace_config.default_catalog)` logic into separate method - e.g. `self._destination_catalog(table)`, as the logic there will grow over time.